app/src/main/java/org/wikipedia/main/MainFragment.java

```diff
@@ -159,11 +193,59 @@ public void onActivityResult(int requestCode, int resultCode, final Intent data)
         } else if (requestCode == Constants.ACTIVITY_REQUEST_LOGIN
                 && resultCode == LoginActivity.RESULT_LOGIN_SUCCESS) {
             FeedbackUtil.showMessage(this, R.string.login_success_toast);
-        } else {
+        } else if (requestCode == ACTIVITY_REQUEST_IMAGE_SEARCH) {
+             final boolean isCamera;
+                if (data == null || data.getData() == null) {
+                    isCamera = true;
+                }
+                else {
+                    final String action = data.getAction();
+                    if (action == null) {
+                        isCamera = false;
+                    }
+                    else {
+                    isCamera = action.equals(android.provider.MediaStore.ACTION_IMAGE_CAPTURE);
+                    }
+                }
+            Uri selectedImageUri;
+            if (isCamera) {
+                selectedImageUri = outputFileUri;
+            } else {
+                selectedImageUri = data == null ? null : data.getData();
+            }
+            if (selectedImageUri != null) {
+                findImageLabels(selectedImageUri); //After this is complete, imageLabels is populated
+            } else {
+                Toast.makeText(getActivity(), "No image was selected", Toast.LENGTH_SHORT).show();
+            }
+        }
+        else {
             super.onActivityResult(requestCode, resultCode, data);
         }
     }
 
+    //This asynchronously calls the ImageLabeler to recognize the image and populates imageLabels with relevant labels
+    private void findImageLabels(Uri selectedImageUri) {
+        Toast.makeText(getActivity(), "Recognizing image...", Toast.LENGTH_SHORT).show();
+        onImageSearchProgressBar(true);
+        String encodedImage = imageEncoder.encodeUriToBase64Binary(getContext(), selectedImageUri);
+        updateImageLabels(getContext(), encodedImage);
+    }
+
+    private void updateImageLabels(Context context, String encodedImage) {
+        CallbackTask.execute(() -> getLabelList(encodedImage), new CallbackTask.DefaultCallback<List <String>>() {
+            @Override
+            public void success(List <String> list) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: ['<' is preceded with whitespace.](https://app.codacy.com/app/amawai/android-wikipedia-390/pullRequest?prid=1541123)